x-pack/plugin/security/src/main/java/org/elasticsearch/xpack/security/rest/action/user/RestChangePasswordAction.java

```diff
@@ -61,7 +62,8 @@ public RestChannelConsumer innerPrepareRequest(RestRequest request, NodeClient c
         final String refresh = request.param("refresh");
         return channel ->
                 new SecurityClient(client)
-                        .prepareChangePassword(username, request.requiredContent(), request.getXContentType())
+                    .prepareChangePassword(username, request.requiredContent(), request.getXContentType(), XPackSettings
+                        .PASSWORD_HASHING_ALGORITHM.get(settings))
```

Comment:
I suggest we resolve this once in the ctr, and pass down the enum.

Reply:
Agreed, will do the same for `RestPutUserAction`